### PR TITLE
chore: Include volumeSize in Instancetype hash calculations for `v0.35.x`

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Since volumeSize is not hashed included https://github.com/aws/karpenter-provider-aws/issues/5447. Karpenter will hash the volumeSize as a string 

**How was this change tested?**
- /karpenter snapshot

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates 
- [ ] Yes, issue opened: # 
- [x] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by engedaam — COMMENTED** [review]
/karpenter snapshot

File: pkg/providers/instancetype/instancetype.go [hunk 1]
```diff
@@ -113,7 +113,11 @@ func (p *Provider) List(ctx context.Context, kc *corev1beta1.KubeletConfiguratio
 	subnetHash, _ := hashstructure.Hash(subnets, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
 	kcHash, _ := hashstructure.Hash(kc, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
 	blockDeviceMappingsHash, _ := hashstructure.Hash(nodeClass.Spec.BlockDeviceMappings, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
-	key := fmt.Sprintf("%d-%d-%d-%016x-%016x-%016x-%s-%s", p.instanceTypesSeqNum, p.instanceTypeOfferingsSeqNum, p.unavailableOfferings.SeqNum, subnetHash, kcHash, blockDeviceMappingsHash, aws.StringValue((*string)(nodeClass.Spec.InstanceStorePolicy)), aws.StringValue(nodeClass.Spec.AMIFamily))
+	VolumeSizeStringCollection := lo.Reduce(nodeClass.Spec.BlockDeviceMappings, func(agg string, block *v1beta1.BlockDeviceMapping, _ int) string {
+		return fmt.Sprintf("%s/%s", agg, block.EBS.VolumeSize)
```
Comment (jonathan-innis) [inline pkg/providers/instancetype/instancetype.go:117]: Is it possible that EBS isn't set here and this causes a panic? I can't remember what we require in our API

File: pkg/providers/instancetype/instancetype.go [hunk 2]
```diff
@@ -113,7 +113,11 @@ func (p *Provider) List(ctx context.Context, kc *corev1beta1.KubeletConfiguratio
 	subnetHash, _ := hashstructure.Hash(subnets, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
 	kcHash, _ := hashstructure.Hash(kc, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
 	blockDeviceMappingsHash, _ := hashstructure.Hash(nodeClass.Spec.BlockDeviceMappings, hashstructure.FormatV2, &hashstructure.HashOptions{SlicesAsSets: true})
-	key := fmt.Sprintf("%d-%d-%d-%016x-%016x-%016x-%s-%s", p.instanceTypesSeqNum, p.instanceTypeOfferingsSeqNum, p.unavailableOfferings.SeqNum, subnetHash, kcHash, blockDeviceMappingsHash, aws.StringValue((*string)(nodeClass.Spec.InstanceStorePolicy)), aws.StringValue(nodeClass.Spec.AMIFamily))
+	VolumeSizeStringCollection := lo.Reduce(nodeClass.Spec.BlockDeviceMappings, func(agg string, block *v1beta1.BlockDeviceMapping, _ int) string {
```
Comment (jonathan-innis) [inline pkg/providers/instancetype/instancetype.go:116]: TODO statement around removing this when we get the update into the code around drift hash versioning

Comment (jonathan-innis) [reply pkg/providers/instancetype/instancetype.go:116]: Maybe also a statement around _why_ we are making this change.

File: pkg/providers/instancetype/instancetype.go (hunk 1 above)
Comment (engedaam) [reply pkg/providers/instancetype/instancetype.go:117]: EBS is a required field https://github.com/aws/karpenter-provider-aws/blob/f5241ea49180d2392822fe7750c0a7d9b29aea3f/pkg/apis/v1beta1/ec2nodeclass.go#L234

**Pull request closed without merging**